cmd/astro-wasm/astro-wasm.go

```diff
@@ -133,6 +152,36 @@ func preprocessStyle(i int, style *astro.Node, transformOptions transform.Transf
 	style.FirstChild.Data = str
 }
 
+func Parse() interface{} {
+	return js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		source := jsString(args[0])
+		parseOptions := makeParseOptions(js.Value(args[1]))
+
+		var doc *astro.Node
+		nodes, err := astro.ParseFragment(strings.NewReader(source), &astro.Node{
+			Type:     astro.ElementNode,
+			Data:     atom.Template.String(),
+			DataAtom: atom.Template,
+		})
+		if err != nil {
+			fmt.Println(err)
+		}
+		doc = &astro.Node{
+			Type: astro.DocumentNode,
+		}
+		for i := 0; i < len(nodes); i++ {
+			n := nodes[i]
+			doc.AppendChild(n)
+		}
+
+		result := printer.PrintToJSON(source, doc, parseOptions)
```

Comment:
This is very similar to the `Transform` function other than the way we print. Instead of generating a JS module from our AST, we have a new printer that generates stringified JSON.